src/wallet.ts

```diff
@@ -124,3 +125,52 @@ export class Wallet extends AdapterL2(AdapterL1(ethers.Wallet)) {
         return await this.provider.broadcastTransaction(await this.signTransaction(populatedTx));
     }
 }
+
+interface SigningFunction {
+    (this: AbstractWallet, transaction: TransactionRequest): Promise<string>;
```

Comment:
does it need access to `AbstractWallet`?

Reply:
It could otherwise accept the wallet as a parameter as: 
```ts
    (abstractWallet: AbstractWallet, transaction: TransactionRequest): Promise<string>;
```